pkg/operator/controller/limit_checker.go

```diff
@@ -36,9 +36,14 @@ func (c *checker) Check(ctx context.Context) (bool, int, error) {
 
 func (c *checker) countScanJobs(ctx context.Context) (int, error) {
 	var scanJobs batchv1.JobList
-	err := c.client.List(ctx, &scanJobs, client.MatchingLabels{
+	listOptions := []client.ListOption{client.MatchingLabels{
 		starboard.LabelK8SAppManagedBy: starboard.AppStarboard,
-	}, client.InNamespace(c.config.Namespace))
+	}}
+	if !c.config.ScanJobInSameNamespace {
```

Comment:
The same instance of LimitChecker is use by other controllers, whereas `config.ScanJobInSameNamespace` is only pertinent to vulnerability scans. (We should probably rename it to `config.VulnerabilityScannerSameNamespace` to make it more explicit.)

One solution might be to create another instance of LimitCheck that's aware of such config and inject it into VulnerabilityReportReconciler.

Reply:
As discussed, this checker will check in entire cluster when `samenamespace` option is enabled. So I think this code should work fine.